src/commands.cc

```diff
@@ -239,6 +239,9 @@ void write_buffer(const ParametersParser& parser, Context& context, const ShellC
     if (parser.positional_count() == 0 and !(buffer.flags() & Buffer::Flags::File))
         throw runtime_error("cannot write a non file buffer without a filename");
 
+    if ((context.buffer().flags() & Buffer::Flags::ReadOnly) and parser.positional_count() == 0)
+        throw runtime_error("cannot overwrite the buffer when in readonly mode");
+
```

Comment:
Do we want to allow writing explicitely to the the buffer existing name ? or do we need to check we are referencing another file ?